tests/test_flows.py

```diff
@@ -2417,6 +2418,54 @@ async def my_hook(flow, flow_run, state):
     return my_hook
 
 
+class TestFlowHooks:
+    def test_hook_with_extra_default_arg(self):
+        from prefect import flow
+
+        def hook(flow, flow_run, state, foo=42):
+            assert hook.__name__ == "hook"
+            assert state.is_completed()
+            assert foo == 42
+
+        @flow(on_completion=[hook])
+        def foo_flow():
+            pass
+
+        foo_flow()
```

Comment:
this test could easily pass without the hook being called because the asserts are hidden in the hook itself; i think you should do something like:

```python
global_dict = {}

       def hook(flow, flow_run, state, foo=42):
            global_dict["name"] = hook.__name__
            global_dict["state"]= state
            global_dict["foo"] = foo
```
and then make asserts on the dictionary in the body of the test